### ISSUE TITLE
Problem in computing softmax

### ISSUE THREAD

**Issue by sinahmr** [dev4488/VAE_gumble_softmax#5]
Hi, thank you for sharing this project.

I think line 108 should be changed from
```return self.decode(z),F.softmax(q)```
to
```return self.decode(z), F.softmax(q_y, dim=-1).view(-1, latent_dim * categorical_dim)```.
Since the softmax should be taken on ```categorical_dim``` sized vectors, and not on ```latent_dim * categorical_dim``` sized ones.
